forge-ai/src/main/java/forge/ai/AiController.java

```diff
@@ -570,11 +576,41 @@ private Card chooseBestLandToPlay(CardCollection landList) {
 
                 nonTappedLands.add(land);
             }
+
+            // if we have the choice, see if we can play an untapped land
             if (!nonTappedLands.isEmpty()) {
-                landList = nonTappedLands;
+                // get the costs of the nonland cards in hand and the mana we have available.
+                // If adding one won't make something new castable, then pick a tapland.
+                int max_inc = 0;
+                for (Card c : nonTappedLands) {
+                    max_inc = max(max_inc, c.getMaxManaProduced());
+                }
+
+                int mana_available = 0;
+                for (Card c : player.getCardsIn(ZoneType.Battlefield)) {
+                    mana_available += c.getMaxManaProduced();
+                }
+
+                boolean found = false;
+                for (Card c : nonLandsInHand) {
+                    // TODO make this work better with split cards and Monocolored Hybrid
+                    if (c.getManaCost().getCMC() == max_inc + mana_available) {
```

Comment:
doesn't this need kind of a sanity check like `max_inc` > 0 so you're not thinking something like _Maze of Ith_ gets you in casting range?

Reply:
Good catch!